src/Comsave/SafeSalesforceSaverBundle/Producer/RpcSfSaverClient.php

```diff
@@ -7,49 +7,55 @@
 use OldSound\RabbitMqBundle\RabbitMq\RpcClient;
 use PhpAmqpLib\Exception\AMQPTimeoutException;
 
-/**
- * Class RpcSfSaverClient
- * @package Comsave\SafeSalesforceSaverBundle\Producer
- */
 class RpcSfSaverClient
 {
+    public const REQUEST_EXPIRATION = 50;
+
     /** @var RpcClient */
     private $rpcClient;
 
     /**
-     * @param RpcClient $rpcClient
      * @codeCoverageIgnore
      */
     public function __construct(RpcClient $rpcClient)
     {
         $this->rpcClient = $rpcClient;
     }
 
-    /**
-     * @param $models
-     * @return mixed
-     * @throws TimeoutException
-     * @throws UnidentifiedMessageException
-     */
-    public function call($models): string
+    public function call(string $serializedModels): string
     {
-        $requestId = 'sss_' . crc32($models);
-        $this->rpcClient->addRequest($models, 'safe_salesforce_saver_server', $requestId, null, 50);
+        $requestId = $this->addRequest($serializedModels);
 
         try {
             $reply = $this->rpcClient->getReplies();
         } catch (AMQPTimeoutException $e) {
-            throw new TimeoutException($models);
+            throw new TimeoutException($serializedModels);
         }
 
         if (!isset($reply[$requestId])) {
-            throw new UnidentifiedMessageException($requestId, $models);
-        }
-
-        if (gettype($reply[$requestId]) != 'string' ) {
-            return serialize($reply[$requestId]);
```

Comment:
@Dirk-Scholten Do you think this will work in all projects?

Reply:
I've disabled the default deserialization behavior so we always handle the serialization/unserialization ourselves.

Reply:
If the defaults are off it should be fine. That would be much better. We just want to receive one or more models to save and then do our own stuff, without any default behaviour getting in the way.